encoding/protobuf/error.go

```diff
@@ -110,12 +103,12 @@ func convertToYARPCError(encoding transport.Encoding, err error) error {
 			return convertErr
 		}
 		detailsBytes, cleanup, marshalErr := marshal(encoding, st.Proto())
+		defer cleanup()
```

Comment:
I think this is nil on errors right? Let's push this after the error check, I think this may cause a panic